helix-core/src/syntax.rs

```diff
@@ -792,25 +802,20 @@ impl Loader {
     pub fn language_config_for_file_name(&self, path: &Path) -> Option<Arc<LanguageConfiguration>> {
         // Find all the language configurations that match this file name
         // or a suffix of the file name.
-        let configuration_id = path
-            .file_name()
-            .and_then(|n| n.to_str())
-            .and_then(|file_name| self.language_config_ids_by_extension.get(file_name))
+        let configuration_id = self
+            .language_config_ids_by_glob
+            .iter()
+            .find_map(|(glob, id)| {
+                if glob.compile_matcher().is_match(path) {
```

Comment:
This is fairly inefficient, I the very least we should precompile the glob here but even if we do that matching every glob in sequence is fairly slow.
 
Instead you cab jse a GlobSet. Matching a globset is not much slow than matching a single Glob but matches all globs at once. You can use  https://docs.rs/globset/latest/globset/struct.GlobSet.html#method.matches)

Reply:
I think using the longest match makes sense as a tiebreaker. For example, if we switch to using globs for filextensions in the future then `*.txt` and `*.todo.txt` (example from above) would both match the same baths but `.todo.txt` should be used because its longer (its matches are more specific). I can imagine something similar for other paths. I can't really think of a way where would would want to use the more general pattern over the more specific one.